ts/components/messages/MessageDetailData.tsx

```diff
@@ -0,0 +1,178 @@
+import I18n from "i18n-js";
+import * as pot from "italia-ts-commons/lib/pot";
+import { Text, View } from "native-base";
+import * as React from "react";
+import { StyleSheet } from "react-native";
+import { CreatedMessageWithContent } from "../../../definitions/backend/CreatedMessageWithContent";
+import { ServicePublic } from "../../../definitions/backend/ServicePublic";
+import { ServiceMetadataState } from "../../store/reducers/content";
+import { PaymentByRptIdState } from "../../store/reducers/entities/payments";
+import customVariables from "../../theme/variables";
+import { format, formatDateAsLocal } from "../../utils/dates";
+import { handleItemOnPress } from "../../utils/url";
+import CopyButtonComponent from "../CopyButtonComponent";
+import BlockButtons, { BlockButtonProps } from "../ui/BlockButtons";
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: customVariables.brandGray,
+    padding: customVariables.contentPadding
+  },
+  row: {
+    flexDirection: "row",
+    justifyContent: "space-between"
+  },
+  flex: {
+    flex: 1
+  }
+});
+
+type Props = Readonly<{
+  message: CreatedMessageWithContent;
+  serviceDetail: pot.Pot<ServicePublic, Error>;
+  serviceMetadata?: ServiceMetadataState;
+  paymentsByRptId?: PaymentByRptIdState;
+  goToServiceDetail?: () => void;
+}>;
+
+/**
+ * A component to render a summary of data about the message and the related service.
+ * If data are available, the user can start a call or send and email to the service
+ */
+class MessageDetailData extends React.PureComponent<Props> {
```

Comment:
This implementation is pretty the same but with a refactoring: c399da887af0e4dab188f86f97a695035e3d06ac
Could it be an improvement?